include/dslash_helper.cuh

```diff
@@ -352,8 +352,7 @@ namespace quda
 
       if (in.Location() == QUDA_CUDA_FIELD_LOCATION) {
         // create comms buffers - need to do this before we grab the dslash constants
-        ColorSpinorField *in_ = const_cast<ColorSpinorField *>(&in);
-        static_cast<cudaColorSpinorField *>(in_)->createComms(nFace, spin_project);
+        const_cast<ColorSpinorField &>(in).createComms(nFace, spin_project);
```

Comment:
would it be worth thinking about `createComms(...) const` and using mutable instead of having to `const_cast` (I know this is unrelated to the spirit of this PR)

Reply:
Completely agree.  I had thought about fixing this aspect, but then we decided this PR won't touch the comms, I figured to leave this.